core/dbt/config/utils.py

```diff
@@ -23,3 +23,25 @@ def parse_cli_yaml_string(var_string: str, cli_option_name: str) -> Dict[str, An
     except (DbtValidationError, OptionNotYamlDictError):
         fire_event(InvalidOptionYAML(option_name=cli_option_name))
         raise
+
+
+def exclusive_primary_alt_value_setting(
+    dictionary: Dict[str, Any], primary: str, alt: str
+) -> None:
+    """Munges in place under the primary the options for the primary and alt values
+
+    Sometimes we allow setting something via TWO keys, but not at the same time. If both the primary
+    key and alt key have values, an error gets raised. If the alt key has values, then we update
+    the dictionary to ensure the primary key contains the values. If neither are set, nothing happens.
+    """
+
+    primary_options = dictionary.get(primary)
+    alt_options = dictionary.get(alt)
+
+    if primary_options and alt_options:
+        raise DbtWarnErrorOptionsError(
```

Comment:
This error indicates to me that the method should only be used in the context of `warn_error_options` configuration, and not more generally. However, it's defined generality in config.utils. Is there somewhere closer to WarnErrorOptions (perhaps on the class itself) that this method would be more appropriate? 

Alternatively, we could raise a more generic error here and have the caller catch + handle it for its specific context.

Reply:
I think we should probably go the more generic error here and have the the caller catch + handle it for its specific context. The question is what existing error to use or what new error to create. How about something like `DbtExclusivePropertyUseError`?

Currently intentionally skip a whole bunch of errors in [read_project_flags](https://github.com/dbt-labs/dbt-core/blob/49b66e10549190128a1a228a41a2d3f31c1f2ef1/core/dbt/config/project.py#L853). I originally wrote `exclusive_primary_alt_value_setting` to raise a `DbtRuntimeError`, but this was explicitly being swallowed by `read_project_flags` exception handling. I then tried `DbtConfigError`, which actually seemed more reasonable and better fitting. However it turns out that `DbtConfigError` is a subclass of `DbtRuntimeError` and thus was being swallowed by the same logic mentioned previously. Additionally if we start raising instead of swallowing `DbtConfigError`s some tests start failing, which means we actually depend on `DbtConfigError` being swallowed here 🙃 

---
Separately, we _could_ put this in `WarnErrorOptions` in `dbt-common`, and if we want to we can. However, I want to push back against doing so:
1. This likely won't be our last instance of changing terminology where we want to use this logic (thus having the generalized implementation is useful)
2. The conceptually the flexibility is on the input end, not on the object end. Thus by implementing the flexibility on the input end, the implementation matches the conceptual mental model.
3. (fuzzier) It's been discussed whether events should be more dynamic (i.e. the level -- `debug`, `info`, `warn`, `error` -- can configured for any event). If we move that way, we'd likely want to create a new object and deprecate `WarnErrorOptions`, and instead of having to duplicate the logic across objects, it exists separately in core prior to instantiation.

Reply:
`DbtExclusivePropertyUseError` sounds good 👍 . We can also pass in the `config_name` (in this case, "warn_error_options") as a parameter to this method for templating the error message